initfiles/bin/init_sasha

```diff
@@ -15,40 +15,19 @@
 #    limitations under the License.
 ################################################################################
 
-PATH_PRE=`type -path hpcc_setenv`
+PATH_PRE=$(type -path hpcc_setenv)
 source ${PATH_PRE}
-PID_NAME="$PID/`basename $PWD`.pid"
+PID_NAME="$PID/$(basename $PWD).pid"
 
-INSTALL_DIR=`dirname ${PATH_PRE}`/..
+INSTALL_DIR="$(dirname ${PATH_PRE})/.."
 source  ${INSTALL_DIR}/etc/init.d/hpcc_common
 
-INIT_PID_NAME="$PID/init_`basename $PWD`.pid"
-echo $$ > $INIT_PID_NAME
-
 export SENTINEL="saserver.sentinel"
 rm -f ${SENTINEL}
 
-
-
-
-which_pidof
-
-killed() {
-    pid=`${PIDOF} saserver`
-    if [ -n "$pid" ]; then
-        sasha server=. action=stop
-        I=1
-        while [ -n "$pid" ] && [ $I -lt 30 ]; do
-            echo "Waiting for saserver to stop"
-            sleep 2
-            pid=`${PIDOF} saserver`
-            let I=I+1
-        
-        done
-        if [ -n "$pid" ]; then
-            kill -9 $pid
-        fi
-    fi
+killed()
+{
+    sasha server=. action=stop
     kill_process ${PID_NAME} saserver 3 ${SENTINEL}
```

Comment:
So before we had a loop here using PIDOF to check and see if the process was still alive.  I'm pretty sure it's needless though.  Now what will happen is the call to "sasha server=. action=stop" will run, and then upon exiting we will run kill_process.  Kill process will do a SIGKILL if it takes us longer than 3 seconds to make sure saserver is actually dead.  Which makes this entire timeout loop unnecessary, since it basically is a non-generic version of what kill_process does.